File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-4.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */
    helper(current,k){
        if(!current) return
        let left=this.helper(current.left,k)
        if(left!=undefined) return left
        k[0]--
        if(!k[0]) return current.val
        let right=this.helper(current.right,k)
        if(right!=undefined) return right
    }
    kthSmallest(root, k) {
        const kArr=[k]
        return this.helper(root,kArr)
    }
}

File: Data Structures & Algorithms/depth-of-binary-tree/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
          return 0
        level=0
        queue=deque()
        queue.append(root)
        while len(queue):
          level+=1
          for i in range(len(queue)):
            cur=queue.popleft()
            if cur.left:
              queue.append(cur.left)
            if cur.right:
              queue.append(cur.right)
        return level